virtual-desktop/src/app/start-url-manager/app2app-args-parser.service.ts

```diff
@@ -17,26 +17,41 @@ export class App2AppArgsParser {
   private startIndex: number;
   private length: number;
   private data: string;
+  private isFirstFullscreenApp: boolean;
 
   constructor() {
-
+    this.isFirstFullscreenApp = true; // Variable used to keep track of if plugin in desktop URL is for standalone mode
   }
 
-  parse(app2app: string): App2AppArgs {
-    this.startIndex = 0;
-    this.length = app2app.length;
-    this.data = app2app;
-    const pluginId = this.getPart();
-    const actionType = this.getPart();
-    const actionMode = this.getPart();
-    const formatter = this.getPart();
-    const contextData = this.getLastPart();
+  parse(app2appArray: string[]): App2AppArgs {
+    let app2appKey = app2appArray[0];
+    let app2app = app2appArray[1];
+    let pluginId, actionType, actionMode, formatter, contextData;
+    switch(app2appKey) {
+      case "pluginId": // Open In New Browser Tab
+        pluginId = app2app;
+        actionType = "launch";
+        actionMode = "create";
+        formatter = "data";
+        contextData = '{"isFirstFullscreenApp":"' + this.isFirstFullscreenApp + '"}';
```

Comment:
The expression serializes `this.isFirstFullscreenApp` as a string e.g. `{"isFirstFullscreenApp":"true"}`. Shouldn't it be a boolean, like this? `{"isFirstFullscreenApp": true}`. I would change the line to 
```typescript
contextData = JSON.stringify({isFirstFullscreenApp: this.isFirstFullscreenApp});
```

Reply:
There wasn't a comparison logic problem, because TS is smart enough to juggle both, but you are right it would be more correct